src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/filter_table.go

```diff
@@ -70,6 +77,15 @@ func (s *filterStatements) selectFilter(
 func (s *filterStatements) insertFilter(
 	ctx context.Context, filter string, localpart string,
 ) (pos string, err error) {
+	var existingFilter string
+
+	// Check if filter already exists in the database
+	err = s.selectFilterByContentStmt.QueryRowContext(ctx,
+		localpart, filter).Scan(&existingFilter)
+	if existingFilter != "" {
+		return existingFilter, err
+	}
+
```

Comment:
We probably want to add a comment to say that there is a race here if two clients try to insert the same filter at the same time, which is fine.